### PR TITLE
Update troubleshooting.md for windows users

### PR DESCRIPTION
**Pull request by pingshunhuangalex**
Include a detailed process of installing hydrogen (fixing node-gyp error) for windows user.
Include up-to-date guide (e.g. command, packages installed) during the installation, and hope this can save people the troubles.

### CODE REVIEW

File: TROUBLESHOOTING.md [hunk 1]
```diff
@@ -1,5 +1,65 @@
...
+  this in virtual environment, but it didn't end up well. Personally, I don't
+  recommend having Python 3 as default and trying to `apm config set python` later
+  to Python 2 as it messed with my `apm -v`, causing python not detected any
+  more.
+- Later if you want to run Python 3 in virtual environment, you can do:
+  ***for Python installer user:***
+  `virtualenv -p <path/to/your/python3> <envname>`
+  ***for Anaconda user:***
+  `conda create -n <envname> python=3.5 anaconda`
+- Install [ZeroMQ](http://zeromq.org/distro:microsoft-windows)
```
Comment (lgeiger) [inline TROUBLESHOOTING.md]: You don't need ZeroMQ installed on windows. It is packaged with https://github.com/JustinTulloss/zeromq.node.

Comment (pingshunhuangalex) [inline TROUBLESHOOTING.md]: Well, there are lots of alternatives... VS2015 and Python 2 may not be necessary either, both C++ components and Python 2 are included in `npm install --global --production windows-build-tools`. 

Also after trying it myself and talking to n-riesco, now I prefer installing python 3 and just set npm config and apm config python to my python 2 directory.

Just tell me how you want me to change it. Thanks.

Comment (pingshunhuangalex) [inline TROUBLESHOOTING.md]: Updated with all the details I have, that's all I got really.

**Comment by pingshunhuangalex** [comment]
Sorry for the spamming, just want to get everything right. Thanks.

**Comment by lgeiger** [comment]
@pingshunhuangalex Thanks a lot for all your work! Any firsthand experience with windows helps a lot.

We are really working towards a better install experience with `zmq-prebuilt`.

**Pull request merged**